common/djangoapps/student/views/login.py

```diff
@@ -386,6 +386,28 @@ def send_reactivation_email_for_user(user):
     return JsonResponse({"success": True})
 
 
+@ensure_csrf_cookie
+def verify_user_password(request):
+    """
+    If the user is logged in and we want to verify that they have submitted the correct password
+    for a major account change (for example, deleting this user account).
+
+    Args:
+        request (HttpRequest): A request object where the password should be included in the POST fields.
+    """
+    # TODO: Use the rate limiting code from above.
+    try:
+        log.info('username: %r', request.user.username)
+        user = authenticate(username=request.user.username, password=request.POST['password'], request=request)
+        if user:
+            return JsonResponse({'success': True})
+        else:
+            return HttpResponseForbidden(_('Password is incorrect.'))
```

Comment:
Until now, we haven't revealed that a user's password is incorrect.  Even in failed login attempts, we keep it ambiguous that either the email or password may be incorrect.
An alternative to this endpoint is to add a password parameter to the "Retire account" endpoint.

Reply:
I can delete this specific text, but it doesn't change the issue where this endpoint is fundamentally only checking passwords and not email + password combinations. If we want to add this to the new 'retire account' endpoint, we will have to add it as an AC to another team's work, since we're not the team working on that endpoint.